opentelemetry-sdk/src/logs/simple_log_processor.rs

```diff
@@ -116,21 +98,11 @@ impl<T: LogExporter> LogProcessor for SimpleLogProcessor<T> {
     }
 
     fn shutdown(&self) -> OTelSdkResult {
-        self.is_shutdown
```

Comment:
This code is not complicated. It's simple enough.

I don't find the `Mutex<()>` approach making things simpler. In fact, from a readability standpoint, the existing `export` code is much better than the one introduced in this PR. If you're using a `Mutex` to guard access to some resource, then it's best to wrap it with `Mutex` instead of having to manually ensure that we acquire a `Mutex`.

Reply:
I don't disagree at all! 
It felt quite awkward to Mutex lock to do `EventEnabled()`, though it did not require mutability! 
Not worried about performance at all here, just does not feel right to do that.

Regd. Rust idiomatic way to prevent multiple export at the same time - I think it'd be better to model export() as requiring mutable self, but we know that will limit us from achieving higher perf when we need it!

If all agree *this* PR is wrong direction, I can abandon it. (Its harmless with/without this PR, as simple processor is just a learning/test purpose component only)